[18.0][MIG] maintenance_location: Migration to 18.0

ypapouin opened the pull request:
Depends on:

- https://github.com/OCA/maintenance/pull/520

Commit:
[11.0][ADD] maintenance_location

Commit:
[11.0][ADD] maintenance_location

Commit:
[12.0][MIG] maintenance_location

Commit:
[12.0][MIG] maintenance_location

olgamarcocb committed:
[MIG] maintenance_location: Migration to 13.0

olgamarcocb committed:
[MIG] maintenance_location: Migration to 13.0

kluna1998 committed:
[IMP] maintenance_location: black, isort, prettier

kluna1998 committed:
[IMP] maintenance_location: black, isort, prettier

kluna1998 committed:
[MIG] maintenance_location: Migration to 14.0

kluna1998 committed:
[MIG] maintenance_location: Migration to 14.0

etobella committed:
[IMP] maintenance_location: Adapt to OCA

etobella committed:
[IMP] maintenance_location: Adapt to OCA

etobella committed:
[IMP] maintenance_location: Add new fields and behavior, remove widget_selection dependency

etobella committed:
[IMP] maintenance_location: Add new fields and behavior, remove widget_selection dependency

Commit:
[UPD] Update maintenance_location.pot

Commit:
[UPD] Update maintenance_location.pot

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (32 of 32 strings)

Translation: maintenance-14.0/maintenance-14.0-maintenance_location
Translate-URL: https://translation.odoo-community.org/projects/maintenance-14-0/maintenance-14-0-maintenance_location/it/

mymage committed:
Translated using Weblate (Italian)

Commit:
[IMP] maintenace_location: pre-commit stuff

Commit:
[IMP] maintenace_location: pre-commit stuff

Commit:
[MIG] maintenance_location: Migration to 16.0

Commit:
[MIG] maintenance_location: Migration to 16.0

Commit:
[UPD] Update maintenance_location.pot

Commit:
[UPD] Update maintenance_location.pot

BhaveshHeliconia committed:
[IMP] maintenance_location: pre-commit auto fixes

BhaveshHeliconia committed:
[IMP] maintenance_location: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] maintenance_location: Migration to 17.0

BhaveshHeliconia committed:
[MIG] maintenance_location: Migration to 17.0

Commit:
[UPD] Update maintenance_location.pot

Commit:
[UPD] Update maintenance_location.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

ypapouin committed:
[IMP] maintenance_location: pre-commit execution

ypapouin committed:
[MIG] maintenance_location: Migration to 18.0

ypapouin committed:
[IMP] maintenance_location: Add location on request report

ypapouin committed:
[IMP] maintenance_location: Add location on request report

ypapouin committed:
[IMP] maintenance_location: Demo data added

ypapouin committed:
[IMP] maintenance_location: Demo data added

ypapouin committed:
[I18N] maintenance_plan: Translation manual update

Fix oca-checks-po:po-duplicate-model-definition

Quan-nhm commented on maintenance_location/models/maintenance_location.py:
```diff
@@ -0,0 +1,52 @@
+    @api.constrains("parent_id")
+    def _check_category_recursion(self):
+        if self._has_cycle():
+            raise ValidationError(_("Error ! You cannot create recursive Locations."))
```
should change from `_` to `self.env._`
it belong to the migration guides of version 18.0
https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-18.0

> You can replace the call to translate a string () with self.env. for getting some performance improvement in some cases. See https://github.com/odoo/odoo/pull/174844.

You should also remove the import `_`

dalonsod commented on maintenance_location/views/maintenance_equipment.xml:
```diff
@@ -0,0 +1,59 @@
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record model="ir.ui.view" id="hr_equipment_view_form">
+        <field name="name">maintenance.equipment.form@maintenance_location</field>
```
Are these changes at `view` names required?

ypapouin replied:
All view names changes are reverted

dalonsod commented on maintenance_location/views/maintenance_location.xml:
```diff
@@ -0,0 +1,61 @@
+        <field name="res_model">maintenance.location</field>
+        <field name="view_mode">list,form</field>
+        <field name="domain">[]</field>
+        <field name="context">{}</field>
```
You could consider adding a `path` for this action

ypapouin replied:
Path added

dalonsod reviewed (changes requested):
Only code review, minor comments

cgarcia-solvos committed:
[IMP] maintenance_location: Add inverse relation and equipment count to locations

Commit:
[UPD] Update maintenance_location.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-17.0/maintenance-17.0-maintenance_location
Translate-URL: https://translation.odoo-community.org/projects/maintenance-17-0/maintenance-17-0-maintenance_location/

cgarcia-solvos commented:
Hello @ypapouin could cherry-pick https://github.com/OCA/maintenance/pull/546 ?

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (34 of 34 strings)

Translation: maintenance-17.0/maintenance-17.0-maintenance_location
Translate-URL: https://translation.odoo-community.org/projects/maintenance-17-0/maintenance-17-0-maintenance_location/it/

ypapouin committed:
[IMP] maintenance_location: pre-commit auto fixes

ypapouin committed:
[MIG] maintenance_location: Migration to 18.0

ypapouin commented:
@cgarcia-solvos migration re-done on current 17.0 head

dalonsod reviewed (approved):
👍

BhaveshHeliconia reviewed (approved):
Code review LGTM!

cgarcia-solvos commented on maintenance_location/models/maintenance_request.py:
```diff
@@ -0,0 +1,21 @@
+    )
+
+    @api.depends("equipment_id")
+    def _compute_equipment_id(self):
```
"_compute_equipment_id" actually computes the "location_id". Could you rename it to "_compute_location_id"?


```suggestion
    def _compute_location_id(self):
```

cgarcia-solvos reviewed (commented):
Code Review. LGTM! Only minor comment

dalonsod commented:
@yvaucher is this ready to be merged? Thanks!

etobella commented on maintenance_location/models/maintenance_request.py:
```diff
@@ -9,13 +9,15 @@ class MaintenanceRequest(models.Model):
 
-    @api.depends("equipment_id")
-    def _compute_equipment_id(self):
+    @api.depends("equipment_id", "equipment_id.location_id")
```
No, it was done this way for a reason. Imagine I have an equipment in location A, I do a task, and it will be assigned to A. A few months after the requests, I move the equipment from A to B, the request should stay as it was, however, with this change, it will move from A to B.

ypapouin replied:
I understand your point, but since changing the location of an equipement should change all open requests, an alternative implementation could be:

```
    @api.depends(
        "equipment_id", "equipment_id.location_id", "stage_id", "stage_id.done"
    )
    def _compute_location_id(self):
        for record in self:
            if record.equipment_id and record.equipment_id.location_id:
                if record.stage_id and record.stage_id.done:
                    # if the request is done, we want keep existing location
                    pass
                else:
                    record.location_id = record.equipment_id.location_id
            else:
                record.location_id = False
```

I know that setting a value for a computed field was mandatory since Odoo 14, but it does not apply for stored values:

<img width="986" height="592" alt="image" src="https://github.com/user-attachments/assets/94997c5b-4217-446a-9b10-73867ffe867d" />

etobella replied:
:+1:

ypapouin replied:
Added with updated tests

etobella commented:
/ocabot merge nobump
